recipes/default.rb

```diff
@@ -24,7 +24,14 @@
 
 if EncryptedAttributesCookbook::Helpers.require_build_essential?(gem_version)
   include_recipe 'build-essential'
+
+  # this is necessary because the build-essential recipe isn't loaded until
+  # after the attempt to compile the gem has occurred
+  package 'build-essential' do
+    action :nothing
+  end.run_action(:install)
```

Comment:
AFAIK this is Debian/Ubuntu specific solution and will break other platforms support. Moreover, I do not think reimplementing the `build-essential` cookbook here would be the way to go.

But the strange thing is that the current code should work because [build-essential installation is configured to be run at compile time](https://github.com/onddo/encrypted_attributes-cookbook/blob/master/attributes/default.rb#L29).

I could not figure out why this does not work for you. I have not been able to reproduce the error using test-kitchen (`kitchen test chef11old-debian-76`). Any ideas? What version of `build-essential` are you using? Are you including it in some unusual way?